lib/elastomer/version_support.rb

```diff
@@ -78,7 +78,20 @@ def strict_boolean(b)
         {enabled: b}
       else
         b
-      end 
+      end
+    end
+
+    # COMPATIBILITY: handle _op_type -> op_type request param conversion for put-if-absent bnehavior
+    # Returns the (possibly mutated) params hash supplied by the caller.
+    #
+    # https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-index_.html#operation-type
+    def op_type(params = {})
+      if es_version_5_x? && params.key?(:_op_type)
```

Comment:
You might need to handle string keys as well I think.

Reply:
Yeah the check is a hack to maintain those internal supported/unsupported lists in VersionSupport, and then `op_type` will be the shim for things like `def params` in the `audit_log` in gh/gh

Reply:
In the instances I saw the param set in gh/gh it looked to be symbols only there but I can update for safety, good call. I have seen symbol and string keys used on the caller side for some of the index mappings too (the mapping shim patch fixes some of these)